src/magellan_firmware/firmware/magellan_controller/Drivetrain.cpp

```diff
@@ -0,0 +1,54 @@
+#include "Drivetrain.h"
+#include "config.h"
+
+Drivetrain::Drivetrain(ros::NodeHandle& nh) :
+        last_commanded_percent_(0),
+        nh_(nh),
+        throttle_pwm_(ESC_PWM),
+        steering_pwm_(SERVO_PWM),
+        steering_angle_msg_(),
+        steering_angle_publisher_("/platform/turning_radius", &steering_angle_msg_) {
+    throttle_pwm_.ConfigLowLimit(THROTTLE_MIN);
+    steering_pwm_.ConfigOffset(STEERING_OFFSET);
+    steering_pwm_.ConfigLowLimit(STEERING_MIN);
+
+    nh.advertise(steering_angle_publisher_);
+}
+
+void Drivetrain::SetThrottlePercent(double percent) {
+    throttle_pwm_.Set(percent);
+
+    if ( fabs(percent) > 0.1 )
+        last_commanded_percent_ = percent;
+}
+
+void Drivetrain::SetSteeringPercent(double percent) {
+    steering_pwm_.Set(percent);
+    steering_angle_msg_.data = GetTurningRadius(percent);
+    steering_angle_publisher_.publish(&steering_angle_msg_);
+}
+
+void Drivetrain::SetSteeringAngle(double angle) {
+    SetSteeringPercent(GetPercentForSteeringAngle(angle));
+}
+
+double Drivetrain::GetSteeringAngleForPercent(double percent) {
+    return percent * M_PI / 6.0;
```

Comment:
The formula here and on line 40 aren’t equivalent. How are these derived?

Reply:
Yeah, this needs to be cleaned up. I'll update

Reply:
To follow up, one was using radians and one was using degrees. Both are in radians now. The number is just from the max angle of the inside wheel in a turn